### DISCUSSION TITLE
Can I use egui to replace an usual HTML GUI site/app made by framework such as React or Vue?

### DISCUSSION THREAD

**Discussion by liufuyang** [emilk/egui#369] [Q&A]
Sorry if this question sounds a bit silly. I know a some usual frontend development process such as using React or Vue, but I am not an expert in html/css type of frontend development. But I am a fan of using and learning Rust.

So let's say if my team or I have a plan to make some simple UI application (viewing some items, pictures or lists and perhaps provide some pages/functions to submit some form/data), I can actually use egui to implement most of the features I need and the performance would be good?

I am not sure what's the best way to ask this question but let me try put it in another way: Say I want to build a super simple version of twitter, or maybe a simple web shop, or perhaps a dating site/app, is egui a good tool for these types of applications? 

Can one also build UI components that can be shared around the team or be used in different pages/projects? 🤔 

Thank you and sorry for asking so many "blurry" questions :)

**Comment by coderedart** [comment]
don't know about egui. but look at  https://yew.rs/ or https://seed-rs.org/ . they are basically trying to be the react/vue of rust.

**Comment by n2** [comment]
Egui is the easiest gui tool I have ever seen, native and web. I often use it to build some small tool software. But for now, it is not suitable for commercial applications, yet. It is buggy, and may have breaking changes in the future. So my advice is wait, and if you have time, contribute to it.
BTW, as an immediate mode GUI library, egui may not be a good choice for applications that contain too many contents like online shopping, and online dating.
Yaws relies very much on macros, and it's macro grammar is very much like HTML/CSS. Have a look at [Iced](https://lib.rs/crates/iced)?

**Reply by liufuyang** [reply]
Thank you for the info. I would for sure like to help if I have time but seems not realistic for now. I am still trying to find time to test out this library. But hopefully it will developer further 👍
